source/agora/common/TransactionPool.d

```diff
@@ -241,6 +276,30 @@ public class TransactionPool
         assert(this.length() == len_prev - count);
         return txs;
     }
+
+    /***************************************************************************
+
+        Check if the input transaction has any double spending input.
+
+        Params:
+            transaction = the transaction to validate
+
+        Returns:
+            true if the transaction has no double spending input.
+
+    ***************************************************************************/
+
+    public bool isValidTransaction (Transaction tx) @trusted
```

Comment:
`@safe` instead of `@trusted`. `@safe` means the compiler will verify all the code inside the function only calls other `@safe` functions.

`@trusted` tells the compiler to ignore the function body, and to make `isValidTransaction` appear as if it's `@safe`.